app/helpers/answer_helper.rb

```diff
@@ -0,0 +1,79 @@
+#E1924 Spring 2019 Addition
+
+module AnswerHelper
+
+  # Delete responses for given questionnaire
+  def self.delete_existing_responses(question_ids, questionnaire_id)
+    response_ids=[]
+    question_ids.each do |question|
+      Answer.where(question_id: question).each do |answer| #For each of the question's answers, log the response_id if in active period
+        response_ids << answer.response_id if self.in_active_period(questionnaire_id, answer)
+      end
+    end
+
+    user_id_to_answers={}
+    response_ids.uniq.each do |response_id| #For each response id in the array, gather map and info about reviewer
+      response_map = Response.find(response_id).response_map
+      reviewer_id = response_map.reviewer_id
+      reviewed_object_id = response_map.reviewed_object_id
+      assignment_name = Assignment.find(reviewed_object_id).name
+      user = Participant.find(reviewer_id).user
+      answers_per_user = Answer.find_by(response_id: response_id).comments
+      #For each response_id, add its info to the dictionary
+      user_id_to_answers[response_id] = [user.email, answers_per_user, user.name, assignment_name] unless user.nil?
```

Comment:
why not store this in a hash
{email: user.email, answers: answers_per_user, name: user.name, assignment_name: assignment_name}
This way you dont have to memorize what index each element is in